File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/relationship/EventWithDrinksAndCustomersDAO.java
package com.jann_luellmann.thekenapp.data.dao.relationship;

import com.jann_luellmann.thekenapp.data.model.relationship.EventWithDrinksAndCustomers;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;
import androidx.room.Transaction;

@Dao
public interface EventWithDrinksAndCustomersDAO {

    @Transaction
    @Query("SELECT * FROM event")
    LiveData<List<EventWithDrinksAndCustomers>> getAll();

    @Transaction
    @Query("SELECT * FROM event WHERE event.name LIKE :name LIMIT 1")
    LiveData<EventWithDrinksAndCustomers> findByName(String name);

    @Transaction
    @Query("SELECT * FROM event WHERE eventId = :id LIMIT 1")
    LiveData<EventWithDrinksAndCustomers> findById(long id);
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/relationship/EventWithDrinksAndCustomersDAO.kt
package com.jann_luellmann.thekenapp.data.dao.relationship

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Query
import androidx.room.Transaction
import com.jann_luellmann.thekenapp.data.model.Event
import com.jann_luellmann.thekenapp.data.model.relationship.EventWithDrinksAndCustomers

@Dao
interface EventWithDrinksAndCustomersDAO {
    @get:Query("SELECT * FROM event")
    @get:Transaction
    val all: LiveData<List<EventWithDrinksAndCustomers>>

    @Transaction
    @Query("SELECT * FROM event WHERE event.ename LIKE :name LIMIT 1")
    fun findByName(name: String): LiveData<EventWithDrinksAndCustomers>

    @Transaction
    @Query("SELECT * FROM event WHERE eventId = :id LIMIT 1")
    fun findById(id: Long): LiveData<EventWithDrinksAndCustomers>

    @Query("DELETE FROM eventCustomerCrossRef WHERE eventId = :eventId")
    fun deleteEventCustomers(eventId: Long)

    @Query("DELETE FROM eventDrinkCrossRef WHERE eventId = :eventId")
    fun deleteEventDrinks(eventId: Long)
}